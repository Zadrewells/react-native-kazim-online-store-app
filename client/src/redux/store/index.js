import createSagaMiddleware from 'redux-saga';
import rootSaga from 'src/redux/sagas';
import createRootReducer from 'src/redux/reducers';

import {
	createStore,
	applyMiddleware,
	compose
} from 'redux';

import {
	isDevelopmentEnv
} from 'src/modules/utils';

const sagaMiddleware = createSagaMiddleware();
const middlewares = [
	sagaMiddleware
];

let composeEnhancer = compose;

if (isDevelopmentEnv()) {
	// composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
}

const store = createStore(
	createRootReducer(),
	{},
	composeEnhancer(
		applyMiddleware(...middlewares)
	)
);

sagaMiddleware.run(rootSaga);

export default store;
