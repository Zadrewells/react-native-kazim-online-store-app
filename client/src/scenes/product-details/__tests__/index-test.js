import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import store from 'src/redux/store';

import {
	Provider
} from 'react-redux';

import {
	ProductDetails
} from 'src/scenes';

describe('<ProductDetails />', () => {

	const getWrapper = () => renderer.create(
		<Provider
			store={store}
		>
			<ProductDetails />
		</Provider>
	).toJSON();

	it('renders correctly', () => {
		const tree = getWrapper();
		expect(tree).toMatchSnapshot();
	});
});
