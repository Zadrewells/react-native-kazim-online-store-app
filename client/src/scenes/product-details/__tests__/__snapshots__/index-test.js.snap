// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetails /> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff0000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
